### PR TITLE
Fix update installplan status conflict

### PR DESCRIPTION
**Pull request by aiyijing**
Signed-off-by: aiyijing <aiyijing@live.com>

**Description of the change:**
Fix update installplan status conflict.
**Motivation for the change:**
Empty installplan will cause the Subscription status to block in UpgradePending
**Architectural changes:**

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

Close https://github.com/operator-framework/operator-lifecycle-manager/issues/2795

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
The PR looks good but need to correct/investigate the unit test failure.

**Comment by perdasilva** [comment]
@aiyijing can you please rebase?

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -2538,3 +2549,45 @@ func (o *Operator) apiresourceFromGVK(gvk schema.GroupVersionKind) (metav1.APIRe
 	logger.Info("couldn't find GVK in api discovery")
 	return metav1.APIResource{}, olmerrors.GroupVersionKindNotFoundError{Group: gvk.Group, Version: gvk.Version, Kind: gvk.Kind}
 }
+
+const (
```
Comment (perdasilva) [inline pkg/controller/operators/catalog/operator.go:2553]: These don't seem to be used anywhere. Needed?

**Review by perdasilva — CHANGES_REQUESTED** [review]
Please remote isSupported and its structures, or use it somewhere

**Comment by perdasilva** [comment]
closing PR as stale. Please re-open if it's still important.

**Pull request closed without merging**